protected_branches.go

```diff
@@ -185,6 +187,34 @@ func (s *ProtectedBranchesService) UnprotectRepositoryBranches(pid interface{},
 	return s.client.Do(req, nil)
 }
 
+// AllowForcePushOptions represents the available
+// AllowForcePush() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/protected_branches.html#allow_force_push # FIXME: not documented yet but this is what is done by the UI
+type AllowForcePushOptions struct {
+	AllowForcePush *bool `url:"allow_force_push,omitempty" json:"allow_force_push,omitempty"`
+}
+
+// AllowForcePush updates the allow force push option.
+//
+// Gitlab API docs:
+// https://docs.gitlab.com/ee/api/protected_branches.html#allow_force_push # FIXME: not documented yet but this is what is done by the UI
+func (s *ProtectedBranchesService) AllowForcePush(pid interface{}, branch string, opt *AllowForcePushOptions, options ...RequestOptionFunc) (*Response, error) {
```

Comment:
Note: this could probably be merged with `RequireCodeOwnerApprovals()` into a generic `UpdateProtectedBranch()` function.

As of today, the documentation is not yet updated to describe  allow_force_push` updates but this is what is done by the UI when the button is toggled:

![image](https://user-images.githubusercontent.com/318490/116991417-ce5a4a00-acd4-11eb-8312-722e6a090ace.png)

Reply:
As this is not (yet) a method that is described in the docs, I prefer to omit it for now and only set this using https://docs.gitlab.com/ee/api/protected_branches.html#protect-repository-branches

Reply:
Ok, I asked if it was intentional or not in the Gitlab MR implementing this https://gitlab.com/gitlab-org/gitlab/-/merge_requests/55261#note_571554927